python/src/iceberg/io/s3.py

```diff
@@ -0,0 +1,111 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Optional
+from urllib.parse import urlparse
+
+import boto3
+
+from iceberg.io.file import FileIO
+
+
+class S3Url(object):
+    """From https://stackoverflow.com/a/42641363
```

Comment:
When we take code from places like Stack Overflow, we need to make sure that we have a license for it and that we document the license and copyright information.

Reply:
The class was overkill anyway, I switched to just using urlparse directly from the standard library.